File: kotlin/src/service/UserService.kt
package com.example.service

import com.example.model.User
import com.example.repository.UserRepository

class UserService(private val repository: UserRepository) {

    lateinit var currentUser: User

    fun getCurrentUser(): User {
        return currentUser!!
    }

    fun login(username: String, password: String): User {
        val user = repository.findByUsername(username)
        return user!!
    }

    fun updateProfile(user: User) {
        repository.save(user)
    }

    fun deleteUser(userId: Long) {
        repository.delete(userId)
    }
}

File: swift/src/UserService.swift
import Foundation

// MARK: - User Service

class UserService {
    private let repository: UserRepository!

    init(repository: UserRepository) {
        self.repository = repository
    }

    func createUser(id: Int, username: String, email: String) -> User {
        let user = User(id: id, username: username, email: email)
        repository.save(user)
        return user
    }

    func getUser(id: Int) -> User! {
        return repository.findById(id)
    }

    func getUserByUsername(_ username: String) -> User! {
        return repository.findByUsername(username)
    }

    func deactivateUser(id: Int) -> Bool {
        guard let user = repository.findById(id) else { return false }
        user.isActive = false
        return true
    }

    func deleteUser(id: Int) {
        repository.delete(id)
    }

    func getActiveUsers() -> [User] {
        let allUsers: [User] = []
        return allUsers.filter { $0.isActive }
    }
}
